az-digital/az_quickstart
# Closes #1938: Prepare Quickstart for Drupal 9.5.x (and 10.x).

## Description

Drupal 9.5.0 will be released this month.  We'd like to update Quickstart to use this new Drupal core minor version as part of our next Quickstart minor release (`2.6.0`) if possible.

We won't want to merge this until the final Drupal 9.5.0 release is available.

### 9.5.x / 10.x specific changes
- Updated Drupal core patches that no longer applied with Drupal core `9.5.x`
- Addressed PHP coding standards issues flagged by updated PHPCS configuration
- Updated JS transpilation dependencies in package.json to match Drupal `9.5.x`
- Re-generated `.js` files from `.es6.js` files using updated JS tools
- Updated version of NodeJS installed by ProboCI to `18.x` (current LTS release)
- Added Drupal 10 to `core_version_requirement` property in `.info.yml` files
- Removed deprecated core modules from list of modules to install by default (color and RDF)
- Added DB update for uninstalling depreecated core modules (color, hal, quickedit, rdf).  Ckeditor (4) and aggregator are also deprecated

### Changes that might need a separate PR so they can be backported
- ~Added steps to `.probo.yml` to install the Upgrade Status module on Probo review sites~ (moved to #2135)
- ~Addressed Twig deprecations (see https://twig.symfony.com/doc/2.x/deprecated.html#tags)~ (moved to #2107)
  > ~Adding an `if` condition on a `for` tag is deprecated in Twig 2.10. Use a `filter` filter or an "if" condition inside the "for" body instead (if your condition depends on a variable updated inside the loop).~ 	
- ~Added explicit access checks to entity queries for https://www.drupal.org/node/3201242 (deprecation introduced in Drupal 9.2.x)~ (moved to #2107)

## Related issues
Closes #1938 

## How to test
General testing to ensure that Quickstart features work as expected and checking to see that no errors show up in the logs or status report.

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
